src/main/java/org/opensearch/security/ssl/SslContextHandler.java

```diff
@@ -0,0 +1,166 @@
+            .collect(Collectors.toSet());
+        return currentCertSignatureSet.equals(newCertSignatureSet);
+    }
+
+    private void validateSubjectDns(final List<Certificate> newCertificates) throws CertificateException {
+        final List<String> currentSubjectDNs = keyMaterialCertificates().map(Certificate::subject).sorted().collect(Collectors.toList());
+        final List<String> newSubjectDNs = keyMaterialCertificates(newCertificates).map(Certificate::subject)
+            .sorted()
+            .collect(Collectors.toList());
+        if (!currentSubjectDNs.equals(newSubjectDNs)) {
+            throw new CertificateException(
+                "New certificates do not have valid Subject DNs. Current Subject DNs "
+                    + currentSubjectDNs
+                    + " new Subject DNs "
+                    + newSubjectDNs
+            );
+        }
+    }
+
+    private void validateIssuerDns(final List<Certificate> newCertificates) throws CertificateException {
+        final List<String> currentIssuerDNs = keyMaterialCertificates().map(Certificate::issuer).sorted().collect(Collectors.toList());
+        final List<String> newIssuerDNs = keyMaterialCertificates(newCertificates).map(Certificate::issuer)
+            .sorted()
+            .collect(Collectors.toList());
+        if (!currentIssuerDNs.equals(newIssuerDNs)) {
+            throw new CertificateException(
+                "New certificates do not have valid Issuer DNs. Current Issuer DNs: "
+                    + currentIssuerDNs
+                    + " new Issuer DNs: "
+                    + newIssuerDNs
+            );
+        }
+    }
+
+    private void validateSans(final List<Certificate> newCertificates) throws CertificateException {
+        final List<String> currentSans = keyMaterialCertificates().map(Certificate::subjectAlternativeNames)
+            .sorted()
+            .collect(Collectors.toList());
+        final List<String> newSans = keyMaterialCertificates(newCertificates).map(Certificate::subjectAlternativeNames)
+            .sorted()
+            .collect(Collectors.toList());
+        if (!currentSans.equals(newSans)) {
+            throw new CertificateException(
+                "New certificates do not have valid SANs. Current SANs: " + currentSans + " new SANs: " + newSans
+            );
+        }
+    }
+
+    private void validateNewCertificates(final List<Certificate> newCertificates) throws CertificateException {
+        for (final var certificate : newCertificates) {
+            certificate.x509Certificate().checkValidity();
+        }
+        validateSubjectDns(newCertificates);
+        validateIssuerDns(newCertificates);
+        validateSans(newCertificates);
+    }
+
+    private void invalidateSessions() {
+        final var sessionContext = sslContext.sessionContext();
+        if (sessionContext != null) {
+            final var sessionsIds = sessionContext.getIds();
```

Comment:
What is a session here exactly?

Reply:
Yes current solution does not invalidate open SSL sessions.